modules/ROOT/pages/_partials/sync/server-auth.adoc

```diff
@@ -51,6 +51,29 @@ const server = VoyagerServer(apolloConfig, voyagerConfig)
 
 The link:https://github.com/aerogear/voyager-server/blob/master/examples/keycloak[Keycloak Example Server Guide] has an example server based off the instructions above and shows all of the steps needed to get it running.
 
+= Protecting Subscriptions with Keycloak
+
+If you have set up a subscription server as described in xref:sync-server-realtime-updates[Realtime Updates] and you have set up {keycloak-service} as described above then your application should make use of the `onConnect` function to validate the token provided by the client. This can be done as shown in the example below:
+
+```
+server.applyMiddleware({ app })
+  httpServer.listen({ 4000,
+}, () => {
+  new SubscriptionServer ({
+    execute,
+    subscribe,
+    schema: server.schema,
+    // add the onConnect method to check the client's token
+    onConnect: async connectionParams => {
+      return await keycloakService.validateToken(connectionParams)
```

Comment:
It is important that we read all code samples line by line. 
In this case `keycloakService` was not defined and as reader this will confuse me a lot.

I personally prefer to have smaller code samples with something like:
```
onConnect: async connectionParams => {
      return await keycloakService.validateToken(connectionParams)
}
```
And then explain what this does and where needs to be put

Reply:
@wtrocki based on other comment I removed excess code and l stated above where the keycloakService was set up. But the way you have suggested is better and I will remove almost everything from this code snippet.